platform/macos/display_server_macos.mm

```diff
@@ -3730,6 +3730,12 @@
 #endif
 
 	screen_set_keep_on(GLOBAL_GET("display/window/energy_saving/keep_screen_on"));
+
+	if (!Engine::get_singleton()->is_editor_hint() && !OS::get_singleton()->is_in_low_processor_usage_mode()) {
+		// This is needed to make sure that background work does not starve the main thread.
+		// This is only setting the priority of this thread, not the whole process.
+		pthread_set_qos_class_self_np(QOS_CLASS_USER_INTERACTIVE, 0);
+	}
```

Comment:
This shouldn't do anything, it's the main thread of the GUI app, it should be user-interactive already.

> The main thread is automatically assigned a QoS based on its environment. In an app, the main thread runs at a QoS level of user-interactive.
>
> _(from Energy Efficiency Guide for Mac Apps)_